lib/hsa/mcwamp_hsa.cpp

```diff
@@ -4995,6 +5011,8 @@ HSABarrier::enqueueAsync(hc::memory_scope fenceScope) {
         hsaQueue()->releaseLockedRocrQueue();
```

Comment:
Since you lock the rocr queue earlier in this function, could you remove the locking/unlocking of the queue within this scope?